src/store/memory/subscriber_store.rs

```diff
@@ -32,8 +32,9 @@ impl SubscriberStore for InMemorySubscriberStore {
         Ok(self.subscribers.values().cloned().collect())
     }
 
-    async fn delete(&mut self, id: i32) -> Result<()> {
-        self.subscribers.remove(&id);
+    async fn delete(&mut self, email: &Email) -> Result<()> {
+        self.subscribers
+            .retain(|_key, subscriber| subscriber.email.ne(email));
```

Comment:
I normally see just `_` instead of a variable name when ignoring something inside of a function like this. You don't have to change it, but just thought I'd mention.